File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/StateMachineWorkflowService/VB/Service/ICalculator.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ServiceModel

Namespace Microsoft.WorkflowServices.Samples
	<ServiceContract([Namespace] := "http://Microsoft.WorkflowServices.Samples")> _
	Public Interface ICalculator
		<OperationContract()> _
		Function PowerOn() As Integer
		<OperationContract()> _
		Function Add(ByVal value As Integer) As Integer
		<OperationContract()> _
		Function Subtract(ByVal value As Integer) As Integer
		<OperationContract()> _
		Function Multiply(ByVal value As Integer) As Integer
		<OperationContract()> _
		Function Divide(ByVal value As Integer) As Integer
		<OperationContract()> _
		Sub PowerOff()
	End Interface
End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/StateMachineWorkflowService/CS/Service/ICalculator.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ServiceModel;

namespace Microsoft.WorkflowServices.Samples
{
    [ServiceContract(Namespace = "http://Microsoft.WorkflowServices.Samples")]
    public interface ICalculator
    {
        [OperationContract()]
        int PowerOn();
        [OperationContract()]
        int Add(int value);
        [OperationContract()]
        int Subtract(int value);
        [OperationContract()]
        int Multiply(int value);
        [OperationContract()]
        int Divide(int value);
        [OperationContract()]
        void PowerOff();
    }  
}
